client/client.go

```diff
@@ -431,9 +441,24 @@ type MetricsTimer = metrics.Timer
 // MetricsNopHandler is a noop handler that does nothing with the metrics.
 var MetricsNopHandler = metrics.NopHandler
 
+// Dial creates an instance of a workflow client. This will attempt to connect
+// to the server eagerly and will return an error if the server is not
+// available.
+func Dial(options Options) (Client, error) {
```

Comment:
Curous if you considered taking this opportunity to move to function-style optional args?

Reply:
For most of the Go SDK, the struct options is good enough to still give us future proofing. There's little benefit of a `func Dial(options ...DialOption) (Client, error)` over struct-based if you 1) have no other params, 2) don't need to preserve backwards compatibility, and 3) can reasonably copy the struct.

I could be leaving off an important benefit, so if there is a good enough argument I'd definitely reconsider.